### PR TITLE
Kelsha Darby - User Stories 39 & 41

### LINKED ISSUE #42: User Story 39, Disabled Merchant Item's are inactive
As an admin
When I visit the merchant index page
And I click on the "disable" button for an enabled merchant
Then all of that merchant's items should be deactivated

### LINKED ISSUE #44: User Story 41, Enabled Merchant Item's are active
As an admin
When I visit the merchant index page
And I click on the "enable" button for a disabled merchant
Then all of that merchant's items should be activated

### PR DESCRIPTION
**Pull request by kelshadarby**
## User Story 39 - Disabled Merchant Items Are Inactive
Closes #42 

**_Test_**
- Wrote `spec/features/admin/merchants/index_spec.rb` to check that once a merchant is disabled that merchants items are inactive
- Added to `spec/model/item_spec` to test the `deactivate` model method

**_Controller_**
- Added to merchant update method to deactivate a merchants items when the merchant is disabled

**_Model_**
- Created the `deactivate` method that changes the items `active?` column to false

## User Story 41 - Enabled Merchant Items Are Active
Closes #44 

**_Test_**
- Wrote `spec/features/admin/merchants/index_spec.rb` to check that once a disabled merchant is enabled that merchants items are active
- Added to `spec/model/item_spec` to test the `activate` model method

**_Controller_**
- Added to merchant update method to activate a merchants items when the merchant is enabled

**_Model_**
- Created the `activate` method that changes the items `active?` column to true

### CODE REVIEW

File: app/views/items/index.html.erb [hunk 1]
```diff
@@ -13,8 +13,6 @@
       <p> <%= item.description unless @merchant%> </p>
       <p>Price: <%=number_to_currency(item.price) %> </p>
       <p>Inventory: <%= item.inventory %> </p>
-      <% if !@merchant %>
-      <% end %>
```
Comment (fredrondina) [inline app/views/items/index.html.erb:17]: nice job cleaning this up

File: spec/features/admin/merchant/index_spec.rb [hunk 2]
```diff
@@ -78,6 +78,32 @@
...
 
+    it "All items from a disabled merchant are inactive" do
+      visit '/admin/merchants'
+
+      within "#merchant-#{@meg.id}" do
+        click_button "Disable"
+      end
+
+      @meg.items.each do |item|
+        expect(item.active?).to eq(false)
```
Comment (fredrondina) [inline spec/features/admin/merchant/index_spec.rb:89]: nice test

File: app/controllers/admin/merchants_controller.rb [hunk 3]
```diff
@@ -7,9 +7,11 @@ def update
     merchant = Merchant.find(params[:merchant_id])
     if merchant.active?
       merchant.disable
+      merchant.items.each { |item| item.deactivate }
```
Comment (brian-greeson) [inline app/controllers/admin/merchants_controller.rb]: Should this live inside the merchant model?  It seems to be a part of the merchant.disable method

File: app/models/item.rb [hunk 4]
```diff
@@ -25,15 +25,23 @@ def no_orders?
     item_orders.empty?
   end
 
-  def self.most_popular(num_items_to_show)      
+  def activate
```
Comment (brian-greeson) [inline app/models/item.rb:28]: easy to understand method names

File: spec/models/item_spec.rb [hunk 5]
```diff
@@ -97,6 +97,21 @@
       expect(Item.least_popular(5)).to eq([tire3, tire4, tire5, tire6, tire1])
     end
 
+    after(:each) do
```
Comment (fredrondina) [inline spec/models/item_spec.rb:125]: does this need to be in here twice? looks like the same thing is done again on lines 146-150

**Review by fredrondina — APPROVED** [review]
Looks good, I would just double check there are two teardowns in the item_spec test

**Review by brian-greeson — CHANGES_REQUESTED** [review]
I think line 10 in the merchant controller `merchant.items.each { |item| item.deactivate }` is supposed to be in the merchant model?

**Review by brian-greeson — APPROVED** [review]
Epic!

**Pull request merged**